internal/controller/controlplaneauth/controller.go

```diff
@@ -133,16 +133,25 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, errors.New(errNotControlPlaneAuth)
 	}
 
+	if meta.WasDeleted(cr) && cr.Spec.ForProvider.TokenSecretRef != nil {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
 	if meta.GetExternalName(cr) == "" {
 		return managed.ExternalObservation{}, nil
 	}
-	uid, err := uuid.Parse(meta.GetExternalName(cr))
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(err, fmt.Sprintf("failed to parse external name as UUID %s", meta.GetExternalName(cr)))
-	}
-	_, err = c.tokens.Get(ctx, uid)
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(uperrors.IsNotFound, err), "failed to get token")
+
+	if cr.Spec.ForProvider.TokenSecretRef == nil {
+		uid, err := uuid.Parse(meta.GetExternalName(cr))
```

Comment:
FYI - we don't place guarantees internally on this ID being stable. i.e. if we do a restore of the control plane, the id will change.

Reply:
it's the uid from an PAT

Reply:
Ahh kk. 👍